src/Widgets/calendar/CalendarModel.vala

```diff
@@ -264,92 +300,120 @@ namespace DateTime.Widgets {
+                E.SourceCalendar cal = (E.SourceCalendar)source.get_extension (E.SOURCE_EXTENSION_CALENDAR);
+                var map = source_events.get (source);
+                if (map != null) {
+                    foreach (var comp in source_events.get (source).values.read_only_view) {
+                        unowned iCal.Component ical = comp.get_icalcomponent ();
+                        foreach (var dt_range in Util.event_date_ranges (ical, data_range)) {
+                            if (dt_range.contains (date)) {
+                                if (!events.has_key (ical.get_uid ())) {
+                                    if (cal.selected == true && source.enabled == true) {
+                                        events.set (ical.get_uid (), new Event (date, dt_range, ical, source));
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            });
 
-        public void change_year (int relative) {
-            month_start = month_start.add_years (relative);
+            var list = new Gee.ArrayList<Event>.wrap (events.values.to_array ());
+            list.sort (sort_events);
+            return list;
         }
 
-        /* --- Helper Methods ---// */
+        //--- Helper Methods ---//
+        public int sort_events (Event e1, Event e2) {
+            if (e1.start_time.compare (e2.start_time) != 0)
+                return e1.start_time.compare(e2.start_time);
+
+            // If they have the same date, sort them wholeday first
+            if (e1.day_event)
+                return -1;
+            if (e2.day_event)
+                return 1;
+            return 0;
+        }
 
         private void compute_ranges () {
             var month_end = month_start.add_full (0, 1, -1);
             month_range = new Util.DateRange (month_start, month_end);
 
-            int dow = month_start.get_day_of_week ();
-            int wso = (int)week_starts_on;
+            int dow = month_start.get_day_of_week();
+            int wso = (int) week_starts_on;
             int offset = 0;
 
-            if (wso < dow) {
+            if (wso < dow)
                 offset = dow - wso;
-            } else if (wso > dow) {
+            else if (wso > dow)
                 offset = 7 + dow - wso;
-            }
 
             var data_range_first = month_start.add_days (-offset);
 
-            dow = month_end.get_day_of_week ();
-            wso = (int)(week_starts_on + 6);
+            dow = month_end.get_day_of_week();
+            wso = (int) (week_starts_on + 6);
 
-            /* WSO must be between 1 and 7 */
-            if (wso > 7) {
+            // WSO must be between 1 and 7
+            if (wso > 7)
                 wso = wso - 7;
-            }
 
             offset = 0;
 
-            if (wso < dow) {
+            if (wso < dow)
```

Comment:
There are a lot of code style regression here. Is this a copy/paste from something?